MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m

```diff
@@ -383,7 +383,11 @@ - (void)openURLInEmbeddedSafari:(NSURL *)url fromClass:(Class)clazz {
 
 - (void)openURLInSafariApp:(NSURL *)url {
   MSLogDebug([MSDistribute logTag], @"Using Safari browser to open URL: %@", url);
-  [MSUtil sharedAppOpenUrl:url options:@{} completionHandler:nil];
+  
+  /* Dispatch the open url call to the next loop to avoid freezing the App new instance start up */
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [MSUtil sharedAppOpenUrl:url options:@{} completionHandler:nil];
```

Comment:
Isn't it better to have it in `sharedAppOpenUrl`?
What if it is called from main thread? You don't need to check current thread?